### ISSUE TITLE
Commit Preview pane isn't closed after committing

### ISSUE THREAD

**Issue by simurai** [atom/github#1812]
### Prerequisites

* Have you tried to reproduce the problem in [Safe Mode][safe-mode]?
* Have you followed all applicable steps in the [debugging guide][debugging-guide]?
* Have you checked the FAQs on the [message board][message-board] for common solutions?
* Have you checked that your issue [isn't already filed][already-filed]?

[safe-mode]: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
[debugging-guide]: http://flight-manual.atom.io/hacking-atom/sections/debugging/
[message-board]: https://discuss.atom.io/c/faq
[already-filed]: https://github.com/atom/github/issues

### Description

The "Commit Preview" shows "No changes to display" once a commit is made.

![screen shot 2018-11-28 at 3 44 42 pm](https://user-images.githubusercontent.com/378023/49133773-a22c8900-f324-11e8-9f2c-44817142ca78.png)

### Steps to Reproduce

1. Open the "Commit Preview" by clicking on the "See All Staged Changes" button.
2. Make a commit.
3. Observe the "Commit Preview" pane

**Expected behavior:**

"Commit Preview" pane is closed

**Actual behavior:**

Pane shows a less useful "No changes to display" message.

**Reproduces how often:**

100%

### Versions

```
Atom    : 1.34.0-dev-9be188141
Electron: 2.0.14
Chrome  : 61.0.3163.100
Node    : 8.9.3

apm  2.1.3
npm  6.2.0
node 8.9.3 x64
atom 1.34.0-dev-9be188141
python 2.7.10
git 2.12.2
```

### Additional Information

An alternative to closing the pane:

- "Show the commit" - @kuychaco, depends on https://github.com/atom/github/pull/1807
- Show the diff for unstaged changes (if any exist)

**Comment by vanessayuenn** [comment]
@simurai if you leave the file patch pane open when you stage or unstage a file, the pane also remains open and shows "No changes to display" (see screenshot). So the commit preview pane behaviour is actually consistent with the rest. I agree though, that the "No changes to display" message is not at all helpful. I think if we were to change it, we should probably change the rest also for consistency sake.  

![image](https://user-images.githubusercontent.com/6842965/49399908-ec59a400-f742-11e8-8728-8c51739ba256.png)

**Comment by simurai** [comment]
> if you leave the file patch pane open when you stage or unstage a file, the pane also remains open and shows "No changes to display" (see screenshot).

Good point! I can't remember the exact reason why we don't just close the file patch pane. Probably some edge case thing. :smile:

Let's wait for https://github.com/atom/github/pull/1807 and then we can consider to "Show the commit" instead.

**Comment by smashwilson** [comment]
> I can't remember the exact reason why we don't just close the file patch pane.

If I'm recalling correctly, I believe it was so that you could still use the navigation buttons to jump to the file or to the mirror patch (the opposite staged/unstaged changes for the same file)?
